### PR TITLE
Patch: Nested layers support, external LayerHost support

### PR DESCRIPTION
**Pull request by dzearing**
This change is heavily based on Thomas' PR for portals and nexus stuff. Thank you Thomas! This should unblock a few scenarios that people hit.

Before when you nested a Layer within a layer (e.g. A Callout within a Panel), the nested layer would not render on its own layer but instead inline. This caused unexpected clipping bugs. Now nested layers will render each on their own projected layer.

Also, to support scenarios where the layered content should actually render in a specific region, you can now wrap your content within an explicit LayerHost. Example:

```typescript
<LayerHost>
  <div>Hello!</div>
  <Panel ... />
</LayerHost>

<LayerHost>
  <div>Hello!</div>
  <ContextualMenu ... />
</LayerHost>
```

In the example above, we have 2 LayerHosts. The first has a Panel. This Panel will always render under the ContextualMenu's DOM, regardless of the order of rendering. This enables where, for example, you want to render within a remote document in another window, rather than the default document body.

### CODE REVIEW

**Comment by ThomasMichon** [comment]
Please let me just finish my PR. Consuming applications will need the additional logic for `PortalNexus` and other concepts.

**Comment by dzearing** [comment]
Thomas and I chatted; we resolved to this:

* Get this change checked in; this will unblock a number of buggy scenarios
* Update his PR to use the simplified nomenclature to support additional host targeting scenarios where you wish to target layer projection to a specific host.

File: src/components/Layer/ProjectedLayer.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  defaultRemoteProps: any;
+}
+
+/**
+ * ProjectedLayer is an internal helper component that projects the contents rendered within a Layer. It is created
+ * by the corresponding LayerHost that the originating Layer communicates with.
+ */
+export class ProjectedLayer extends BaseComponent<IProjectedLayerProps, {}> {
+  private _rootElement: HTMLElement;
+  private _remoteProps: any;
```
Comment (dzearing) [inline src/components/Layer/ProjectedLayer.tsx]: div props?

**Comment by cliffkoh** [comment]
Approved 👏

File: src/components/Layer/LayerHost.tsx [hunk 2]
```diff
@@ -1,82 +1,109 @@
...
+ * appending a div at the end, which projects all content wrapped in the Layer components within. Projecting
+ * DOM to the end of the document allows for overlaying and stacking scenarios.
+ *
+ * Normally you do not need to interact directly with LayerHost. If you render Layers within content that isn't
+ * wrapped within a LayerHost, a LayerHost will be created and appended to the end of the document body, where
+ * layer content will then be projected. However in some circumstances you want Layered content to be rendered
+ * in a specific place rather than document body (for example in a popup window or contained within a scrollable
+ * region.) In those cases, wrap the content wihtin a LayerHost:
+ *
+ * <LayerHost>
```
Comment (cliffkoh) [inline src/components/Layer/LayerHost.tsx]: @example above the example.

**Pull request merged**